src/security/Handshake.cc

```diff
@@ -287,9 +280,12 @@ Security::HandshakeParser::parseChangeCipherCpecMessage()
     // We are currently ignoring Change Cipher Spec Protocol messages.
     skipMessage("ChangeCipherCpec msg [fragment]");
 
+    // This is a TLSv1.3 message or a resuming session
+    if (details->tlsSupportedVersion != AnyP::ProtocolVersion(AnyP::PROTO_TLS, 1, 3))
+        resumingSession = true;
+
```

Comment:
For TLS1.2 and earlier, if the session is resumed, after the server hello  message follows the ChangeCipherSpec message. 
If the session is not resumed after the Server Hello follows the server certificates message then (maybe) some other messages and then the ServerHelloDone. 

For TLSv1.3 after the Server Hello follows an empty ChangeCipherSpec message (to emulate the resuming sessions of earlier TLS versions).

Reply:
Please use the following comment for this `if` statement then:

```C++
// In TLS v1.3, a dummy change_cipher_spec implies Middlebox Compatibility Mode
// In earlier TLS versions, change_cipher_spec starts (encrypted) session resumption
```

You can use `ChangeCipherSpec` instead of `change_cipher_spec` if you prefer but please note that the former is misspelled in the official code. I do not know whether it is better to continue using a misspelled version (for consistency sake), fixed all entries (ditto), or use a new, TLS 1.3-biased spelling (there is no ChangeCipherSpec type in TLS v1.3 AFAICT).

Also, here and in other similar contexts, please consider replacing the `!=` comparison with `<` (or equivalent) operator when checking for TLS v1.3: We should probably assume that, by default, later TLS versions will do what the currently-latest version does.